### DISCUSSION TITLE
How do I set the connection interval

### DISCUSSION THREAD

**Discussion by EdwinFairchild** [hbldh/bleak#1074] [Q&A]
I have a server device(microcontroller) that unfortunately has to do a blocking flash erase operation and this timesout my connection and Bleak disconnects. 

When I am using my other microcontroller as a client instead of bleak on my desktop, I can set the connection interval for the client to something tolerable to not have this disconnect happen. How do I set the client connection interval using bleak or perhaps the backend?

**Comment by dlech** [comment]
There are no client APIs to set the connection interval but the server can send a "L2CAP Connection Parameter Update Request" to request to change the connection interval. It may also be possible to do this by including "Connection Interval Range" in the advertising data  or by implementing the "Peripheral Preferred Connection Parameters" characteristic of the GAP service. I've only used the former, not the latter two.
